binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java

```diff
@@ -49,123 +48,130 @@ abstract class SerdeResolverUtils {
 		if (!matchingSerdes.isEmpty()) {
-			return matchingSerdes.get(0);
+			return context.getBean(matchingSerdes.get(0), Serde.class);
 		}
 
-		// We don't attempt to find a matching Serde for type '?'
-		if (targetType.getRawClass() == null) {
-			return null;
+		// Use standard serde for built-in types
+		Serde<?> standardDefaultSerde = getStandardDefaultSerde(genericRawClazz);
+		if (standardDefaultSerde != null) {
+			return standardDefaultSerde;
 		}
 
-		Serde<?> serde = null;
-		Class<?> genericRawClazz = targetType.getRawClass();
-		if (Integer.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.Integer();
+		// Use fallback if specified and not from std defaults (we know from above that type is not std default
+		// so using a fallback that is std default type would not work)
+		if (fallbackSerde != null && !isSerdeFromStandardDefaults(fallbackSerde)) {
+			return fallbackSerde;
 		}
-		else if (Long.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.Long();
+
+		// Use JsonSerde if type is not exactly Object
+		if (!genericRawClazz.isAssignableFrom((Object.class))) {
+			return new JsonSerde<>(genericRawClazz);
 		}
-		else if (Short.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.Short();
+
+		// Finally, just resort to using the fallback
+		return fallbackSerde;
+	}
+
+	private static Serde<?> getStandardDefaultSerde(Class<?> genericRawClazz) {
+		if (Integer.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Integer();
 		}
-		else if (Double.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.Double();
+		if (Long.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Long();
 		}
-		else if (Float.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.Float();
+		if (Short.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Short();
 		}
-		else if (byte[].class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.ByteArray();
+		if (Double.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Double();
 		}
-		else if (String.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.String();
+		if (Float.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.Float();
 		}
-		else if (UUID.class.isAssignableFrom(genericRawClazz)) {
-			serde = Serdes.UUID();
+		if (byte[].class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.ByteArray();
 		}
-		else if (!isSerdeFromStandardDefaults(fallbackSerde)) {
-			// User purposely set a default serde that is not one of the above
-			serde = fallbackSerde;
+		if (String.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.String();
 		}
-		else {
-			// If the type is Object, then skip assigning the JsonSerde and let the fallback mechanism takes precedence.
-			if (!genericRawClazz.isAssignableFrom((Object.class))) {
-				serde = new JsonSerde(genericRawClazz);
-			}
+		if (UUID.class.isAssignableFrom(genericRawClazz)) {
+			return Serdes.UUID();
```

Comment:
How about `Serdes.serdeFrom(Class<T> type)` instead of all of these ifs ?

Then you catch their `IllegalArgumentException` and return `null` (or whatever logic you'd like to do)

Reply:
Yeh I was originally looking in that class to see if there was some built-in methods that would work for us. They do handle one more type than we do IIRC but I will take a scan at using it.

Reply:
Sure! Let it be then as it is! Probably not related to this PR anyway.